### DISCUSSION TITLE
RPCA Integration Explained

### DISCUSSION THREAD

**Discussion by sidvmahesh** [satijalab/seurat#8475] [Q&A]
Hi All,

Quick question regarding the Reciprocal PCA (RPCA) integration protocol. I see that as per [this RPCA vignette](https://satijalab.org/seurat/articles/integration_rpca.html), "When determining anchors between any two datasets using RPCA, we project each dataset into the others PCA space and constrain the anchors by the same mutual neighborhood requirement."

I'm a little confused, because my understanding is that nearest neighbors for a given cell in dataset A (when projected into the top PCs in dataset B) should be identical for the same cell when dataset B is projected onto dataset A's top PCs, right? In other words, what is the need to project the reference onto the query's top PCs **AND** the query onto the reference's top PCs to derive a SNN graph (used to determine anchors)?

An explanation would be really helpful. Thanks in advance!

**Comment by plezar** [comment]
When you project A into B’s PCs you’re measuring distances along B’s axes of variation, whereas projecting B into A’s PCs measures distances along A’s axes. Even for a pair of cells from A and B that are really the same cell (or have identical expression vectors), the projections won't line up. Unless by some miracle both of your datasets share exactly the same PC basis, then the projections won't be the same.
